### PR TITLE
LVRUBY-17

### PR DESCRIPTION
**Pull request by AlinaKovtun**
## JIRA ticket

http://ssu-jira.softserveinc.com/browse/LVRUBY-17

## Code reviewers

- [x] @PidtserkovnyiVitalii 
- [ ] @vikabelous  

## Summary of issue

Admin workspace

## Summary of change

Add admin authorisation
Changed seeds.rb file

## Review checklist

- [x] Rubocop doesn't raise any warnings and offenses
- [x] New features are covered with new specs
- [x] All specs pass

### CODE REVIEW

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **94.76%** covered at **4.51** hits/line

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **94.76%** covered at **4.51** hits/line

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **94.76%** covered at **4.51** hits/line

File: app/models/user.rb [hunk 1]
```diff
@@ -13,4 +13,24 @@ class User < ApplicationRecord
   has_many :events
 
   scope :approved, -> { where(approved: true) }
+
+  def student?
+    roles.pluck(:name).include?('student')
```
Comment (omaik) [inline app/models/user.rb]: first, it is better to use `exists?` to not load all queries into the application `roles.exists?(name: 'student')`

Comment (omaik) [reply app/models/user.rb]: second, there is a lot of duplication, metaprogramming will solve this problem nicely.

```
%w(student expert mentor teacher admin).each do |role|
  define_method "#{role}?" do
     roles.exists?(name: role)
   end
end
```
(I may mistake sowhere)

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **93.5%** covered at **4.38** hits/line

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **96.2%** covered at **4.62** hits/line

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **96.17%** covered at **4.17** hits/line

**Comment by vikabelous** [comment]
**Code Coverage:**
:green_heart: **96.17%** covered at **4.17** hits/line

**Pull request merged**